src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComObject.cs

```diff
@@ -77,10 +81,11 @@ internal ComObject(IIUnknownInterfaceDetailsStrategy interfaceDetailsStrategy, I
         /// </remarks>
         public void FinalRelease()
         {
-            if (UniqueInstance)
+            if (UniqueInstance && Interlocked.CompareExchange(ref _released, 1, 0) == 0)
             {
                 CacheStrategy.Clear(IUnknownStrategy);
                 IUnknownStrategy.Release(_instancePointer);
```

Comment:
If the caller doesn't have any live references to this object, and the JIT/GC decide that after reading _instancePointer the this reference is no longer referenced, is it possible the finalizer could run concurrently with this call to Release? And if that happens, would that be ok?

Reply:
The call to `GC.SuppressFinalize(this)` should keep the `this` reference alive until after `IUnknownStrategy.Release(_instancePointer)` is done executing.

Reply:
Yup, I wasn't seeing the subsequent line when writing that.